### PR TITLE
CI deps update: dpctl 0.16 and dpnp 0.14

### PR DESCRIPTION
**Pull request by Alexsandruss**
Changes:
- Update dpctl to 0.16 and dpnp to 0.14 versions
- Solve additional sklearn test failures due to extra warnings in newer numpy version:
  - Deselect tests related to them
  - Fix for warning condition in L1 linear models

### CODE REVIEW

File: daal4py/sklearn/linear_model/_coordinate_descent.py
Comment (ethanglaser) [inline daal4py/sklearn/linear_model/_coordinate_descent.py:1]: are changes related to PR?

Comment (Alexsandruss) [reply daal4py/sklearn/linear_model/_coordinate_descent.py:1]: They are required for green CI

**Review by ethanglaser — APPROVED** [review]

**Pull request merged**